loki/frontend/fparser.py

```diff
@@ -479,8 +479,15 @@ def visit_Use_Stmt(self, o, **kwargs):
                 # Import symbol attributes from module, if available
                 for k, v in module.symbol_attrs.items():
                     # Don't import private module symbols
-                    if v.private or (module.default_access_spec == "private" and not v.public):
+                    if v.private:
                         continue
+                    else:
+                        if module.default_access_spec == "private":
+                            if k not in module.public_access_spec and not v.public:
+                                continue
+                        else:
+                            if k in module.private_access_spec:
+                                continue
```

Comment:
The else isn't required after a continue.
```suggestion
                    if module.default_access_spec == "private":
                        if k not in module.public_access_spec and not v.public:
                            continue
                    else:
                        if k in module.private_access_spec:
                            continue
```